The openat(3) wrapper was added in upstream commit f5e0a89ab6f0024f3d3bec5fd9cf631676b44f6c
which adds "OS X portability": https://salsa.debian.org/clint/fakeroot/-/commit/f5e0a89ab6f0024f3d3bec5fd9cf631676b44f6c

At the moment, the implementation seems very macOS specific as it
doesn't initialize mode_t when O_CREAT is not in flags, the va_arg()
invocation assume that mode_t is an int, et cetera. Ultimately, the
current openat() implementation can cause segfaults on Alpine.

This has been reported upstream via Mail to Clint Adams. Until this is
addressed upstream we just disable the openat wrapper as it wasn't
available prior to the aforementioned commit anyhow.

--- fakeroot-1.28.orig/libfakeroot.c
+++ fakeroot-1.28/libfakeroot.c
@@ -2604,7 +2604,7 @@
 }
 #endif
 
-#ifdef HAVE_OPENAT
+#if defined(HAVE_OPENAT) && defined(__APPLE__)
 int openat(int dir_fd, const char *pathname, int flags, ...)
 {
 	mode_t mode;
